From 59dccb83c08904cd2bf508f41bf1322746ee4622 Mon Sep 17 00:00:00 2001
From: Shaul Triebitz <shaul.triebitz@intel.com>
Date: Thu, 4 Aug 2022 16:50:18 +0300
Subject: [PATCH] wifi: mac80211: set link BSSID

For an AP interface, set the link BSSID when the link
is initialized.

Signed-off-by: Shaul Triebitz <shaul.triebitz@intel.com>
Signed-off-by: Johannes Berg <johannes.berg@intel.com>
---
 net/mac80211/iface.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/mac80211/iface.c b/net/mac80211/iface.c
index dc44279..925fedd 100644
--- a/net/mac80211/iface.c
+++ b/net/mac80211/iface.c
@@ -418,9 +418,11 @@ static void ieee80211_link_init(struct ieee80211_sub_if_data *sdata,
 		case NL80211_IFTYPE_AP:
 			ether_addr_copy(link_conf->addr,
 					sdata->wdev.links[link_id].addr);
+			link_conf->bssid = link_conf->addr;
 			WARN_ON(!(sdata->wdev.valid_links & BIT(link_id)));
 			break;
 		case NL80211_IFTYPE_STATION:
+			/* station sets the bssid in ieee80211_mgd_setup_link */
 			break;
 		default:
 			WARN_ON(1);
-- 
2.17.1

